test/utils.cpp

```diff
@@ -814,6 +814,22 @@ TEST_CASE("getcwd() returns current directory of the process", "[utils]")
 		REQUIRE(datadir.substr(datadir.length() - subdir.length()) ==
 			subdir);
 	}
+
+	SECTION("Returns empty string if current directory doesn't exist")
+	{
+		// Create a temporary directory, change to it and delete it. This leads
+		// getcwd to fail.
+		TestHelpers::TempDir tempdir;
+
+		const std::string tempdir_path = tempdir.get_path();
+		INFO("tempdir = " << tempdir_path);
+
+		TestHelpers::Chdir chdir(tempdir_path);
+
+		REQUIRE(0 == ::rmdir(tempdir_path.c_str()));
```

Comment:
This line is unnecessary—`TempDir` will remove the directory at the end of the test.

Reply:
I need to remove it so the test works. Is there an equivalent of `drop` in C++? I could use that instead.

Reply:
Ah, sorry, my bad—you do need to remove the directory, that's the point of the test :)

Reply:
No equivalent to `drop` as far as I know. The best you can do is to allocate `TempDir` on heap, and free it before calling `getcwd`:

```c++
const auto tempdir = std::unique_ptr<TestHelpers::TempDir>(new TestHelpers::TempDir());
Chdir chdir(tempdir->get_path());
tempdir.reset();
REQUIRE("" == utils::getcwd());
```

Reply:
I'll keep it as is, since it isn't worth the trouble to allocate on the heap IMO.